libr/util/mem.c

```diff
@@ -108,86 +108,83 @@ src |__________|_________|
 	r_mem_copybits (dst, src, nbits);
 }
 
-R_API ut64 r_mem_get_num(const ut8 *b, int size, int endian) {
-        ut16 n16;
-        ut32 n32;
-        ut64 n64;
-        switch (size) {
-        case 1: return b[0];
-        case 2:
-                r_mem_copyendian ((ut8*)&n16, b, 2, endian);
-		return (ut64)n16;
-        case 4:
-                r_mem_copyendian ((ut8*)&n32, b, 4, endian);
-		return (ut64)n32;
-        case 8:
-                r_mem_copyendian ((ut8*)&n64, b, 8, endian);
-		return (ut64)n64;
-        }
+R_API ut64 r_mem_get_num(const ut8 *b, int size) {
+	// LITTLE ENDIAN is the default for streams
+	switch (size) {
+	case 1:
+		return r_read_le8 (b);
+	case 2:
+		return r_read_le16 (b);
+	case 4:
+		return r_read_le32 (b);
+	case 8:
+		return r_read_le64 (b);
+	}
 	return 0LL;
 }
 
 // TODO: SEE: R_API ut64 r_reg_get_value(RReg *reg, RRegItem *item) { .. dupped code?
-R_API int r_mem_set_num (ut8 *dest, int dest_size, ut64 num, int endian) {
-	int num4;
-	short num2;
+R_API int r_mem_set_num (ut8 *dest, int dest_size, ut64 num) {
+	// LITTLE ENDIAN is the default for streams
 	switch (dest_size) {
-	case 1: dest[0] = (ut8) num;
+	case 1:
+		r_write_le8 (dest, (ut8)(num & UT8_MAX));
 		break;
-	case 2: num2 = (short)num;
-		r_mem_copyendian (dest, (const ut8*)&num2, 2, endian);
+	case 2:
+		r_write_le16 (dest, (ut16)(num & UT16_MAX));
 		break;
-	case 4: num4 = (int)num;
-		r_mem_copyendian (dest, (const ut8*)&num4, 4, endian);
+	case 4:
+		r_write_le32 (dest, (ut32)(num & UT32_MAX));
 		break;
-	case 8: r_mem_copyendian (dest, (const ut8*)&num, 8, endian);
+	case 8:
+		r_write_le64 (dest, num);
 		break;
 	default:
 		return R_FALSE;
 	}
 	return R_TRUE;
 }
 
-/* XXX TODO check and use system endian */
-// TODO: rename to r_mem_swap() */
-R_API void r_mem_copyendian(ut8 *dest, const ut8 *orig, int size, int endian) {
+R_API void r_mem_swaporcopy(ut8* dest, const ut8* src, int len, bool big_endian) {
+	if (big_endian) {
+		r_mem_swapendian(dest, src, len);
+	} else {
+		memcpy(dest, src, len);
+	}
+}
+
+R_API void r_mem_swapendian(ut8 *dest, const ut8 *orig, int size) {
```

Comment:
I think a comment before this function and before `r_mem_swaporcopy` would be good. Just to give an idea of what they do, considering they are API.